statsmodels/tsa/stattools.py

```diff
@@ -1055,6 +1055,115 @@ def arma_order_select_ic(y, max_ar=4, max_ma=2, ic='bic', trend='c',
     return Bunch(**res)
 
 
+def kpss(x, null_hypo="level", lags=None):
+    """
+    Kwiatkowski-Phillips-Schmidt-Shin test for stationarity.
+
+    Computes the Kwiatkowski-Phillips-Schmidt-Shin (KPSS) test for the null
+    hypothesis that x is level or trend stationary.
+
+    Parameters
+    ----------
+    x : array_like, 1d
+        Data series
+    null_hypo : str{"level", "trend"}
+        Indicates the null hypothesis for the KPSS test
+        * "level" : The data is level stationary (default)
+        * "trend" : The data is trend stationary
+    lags : int
+        Indicates the number of lags to be used. If None (default),
+        lags is set to int(12 * (n / 100) ** (1 / 4)), as outlined in
+        Schwert (1989).
+
+    Returns
+    -------
+    kpss_stat : float
+        The KPSS test statistic
+    p_value : float
+        The p-value of the test
+    lags : int
+        The truncation lag parameter
+    flag: int
+        * -1 : the p-value is smaller than the indicated p-value.
+        * 0 : the p-value is the indicated p-value.
+        * 1 : the p-value is greater than the indicated p-value.
+    crit : dict
+        The critical values at 10%, 5%, 2.5% and 1%. Based on
+        Kwiatkowski et al. (1992).
+
+    Notes
+    -----
+    To estimate sigma^2 the Newey-West estimator is used. If lags is None,
+    the truncation lag parameter is set to int(12 * (n / 100) ** (1 / 4)),
+    as outlined in Schwert (1989). The p-values are interpolated from
+    Table 1 of Kwiatkowski et al. (1992).
+
+    Missing values are not handled.
+
+    References
+    ----------
+    D. Kwiatkowski, P. C. B. Phillips, P. Schmidt, and Y. Shin (1992): Testing
+    the Null Hypothesis of Stationarity against the Alternative of a Unit Root.
+    `Journal of Econometrics` 54, 159–178.
+    """
+    nobs = len(x)
+    x = np.asarray(x)
+    hypo = null_hypo.lower()
+
+    # if n != m * n
+    if nobs != x.size:
+        raise ValueError("x of shape {} not understood".format(x.shape))
+
+    if hypo == "trend":
+        # p. 162 Kwiatkowski et al. (1992): y_t = beta * t + r_t + e_t,
+        # where beta is the trend, r_t a random walk and e_t a stationary
+        # error term.
+        resids = OLS(x, add_constant(range(1, nobs + 1))).fit().resid
```

Comment:
I suppose `add_constant` converts the input to a numpy array, and numpy can take iterators as inputs. I was surprised also when I first wrote this, but it worked so I did not botter with it any further.

Will change to `np.arange`.